bin/envconfig

```diff
@@ -0,0 +1,147 @@
+#!/bin/bash
+
+# setup lsstsw environment
+#
+# source this file from your ~/.bashrc or ~/.zshrc
+#
+# relative to <lsstsw>/bin/
+
+print_error() {
+  >&2 echo -e "$@"
+}
+
+usage() {
+  if [[ ! -z "$1" ]]; then
+    print_error "$0: $1"
+  fi
+
+  # note that heredocs are prefixed with tab chars
+  echo "$(cat <<-EOF
+
+		Usage: $0  [-i] [-r] [-b] [-h]
+
+		Specific options:
+        -i          interactive, propose a list of available environment
+        -r REF      environment name
+        -b bXXXX    activate the environment already used for the build with id bXXXX
+        -h          show this message
+
+		EOF
+  )"
+
+  finish=true
+   
+}
+
+parse_args() {
+  local OPTIND
+  local opt
+
+  while getopts "hrib" opt; do
+    case "$opt" in
+    b)
+      shift;
+      BUILD_ID=$1
+      ;;
+    i)
+      INTERACTIVE=true
+      ;;
+    h)
+      usage
+      ;;
+    r)
+      shift;
+      ENVREF=$1
+      ;;
+    *)
+      usage "Unknown option: ${opt}"
+      ;;
+    esac
+  done
+  shift $((OPTIND-1))
+}
+
+##############
+
+INTERACTIVE=false
+ENVREF=""
+BUILD_ID=""
+
+parse_args "$@"
+
+if [[ !$finish ]]; then
+
+
+if [[ -z $ZSH_NAME ]]; then
+  LSSTSW=$(cd "$(dirname "${BASH_SOURCE[0]}")/.."; pwd) || return 1
+  SUFFIX='sh'
+else
+  LSSTSW=$(cd "$(dirname "$0")/.."; pwd) || return 1
+  SUFFIX='zsh'
+fi
+
+if [[ ! -f "$LSSTSW/eups/current/bin/setups.$SUFFIX" ]]; then
+  echo "error: eups not found in $LSSTSW/eups/current" 1>&2
+  echo "  you may need to [re]run bin/deploy to [re]deploy EUPS." 1>&2
+  return
+fi
+
+export PATH="$LSSTSW/miniconda/bin:$PATH"
+export PATH="$LSSTSW/lfs/bin:$PATH"
+export PATH="$LSSTSW/bin:$PATH"
+
+export MANPATH="$LSSTSW/lfs/share/man:"
+
+# need to activate the miniconda base environment to get the correct list of environments
+source activate "${LSSTSW}/miniconda"
```

Comment:
I guess `base` can't be used here?

Reply:
Like above, I think this is part of a bigger picture, like using a new conda distribution.

Reply:
to follow up, why can't this be `conda activate base` ?

Reply:
`bin/deploy` is deploying the environments under the local miniconda. If we use `base` conda, we will not be able to see the miniconda environments.
We can add extra logic in order to do that but, in my opinion, it is not in the scope of this issue, and will complicate the scripting quite a lot, increasing the possibility of errors.
In other words, I prefer to keep it simple, at least for the moment.

Reply:
I'm not sure I follow. If I do:
```
$ source activate `pwd`/miniconda
```
in my shell then I see that my activated environment becomes `base`.  It therefore seems much clearer for us to do `conda activate base` directly.  The only way these might be different is if somehow the path to `conda` is not correct.